libgstc/pygstc/gstc.py

```diff
@@ -215,44 +216,68 @@ def _send_cmd_line(self, cmd_line):
         result : dictionary
             Response from the IPC
         """
-
-        cmd = cmd_line[0]
         try:
+            cmd = cmd_line[0]
             jresult = self._ipc.send(cmd_line, timeout=self._timeout)
             result = json.loads(jresult)
-        except Exception as exception:
-            self._logger.error('%s error: %s' % (cmd,
-                                                 type(exception).__name__))
-            traceback.print_exc()
-            raise GstcError(type(exception).__name__)
-        if result['code'] != 0:
-            self._logger.error('%s error: %s' % (cmd,
-                                                 result['description']))
-            raise GstdError(result['description'])
-        return result
+            if result['code'] != 0:
+                self._logger.error('%s error: %s' % (cmd,
+                                                     result['description']))
+                raise GstcError(result['description'],
+                                GstcErrorCode.GSTC_NOT_FOUND)
+            return result
+        except ConnectionRefusedError as e:
+            raise GstdError("Failed to communicate with GSTD",
```

Comment:
This is a `Gstc` error. The only `Gstd` errors are the ones where the response has a code other than OK.